apps/condo/domains/common/utils/tables.utils.ts

```diff
@@ -0,0 +1,320 @@
+    return function (search) {
+        if (!search) return
+        const date = moment(search)
+        if (!date.isValid()) return
+        return filter(date.startOf('day').toISOString())
+    }
+}
+
+export const getDayLteFilter: (dataIndex: DataIndexType) => FilterType = (dataIndex) => {
+    const filter = getFilter(dataIndex, 'single', 'dateTime', 'lte')
+    return function (search) {
+        if (!search) return
+        const date = moment(search)
+        if (!date.isValid()) return
+        return filter(date.endOf('day').toISOString())
+    }
+}
+
+export const getBooleanFilter: (dataIndex: DataIndexType) => FilterType = (dataIndex) => {
+    return getFilter(dataIndex, 'single', 'boolean', undefined)
+}
+
+export const getSortersFromQuery = (query: ParsedUrlQuery): SorterColumn[] => {
+    let sorters = get(query, 'sort', [])
+    if (!Array.isArray(sorters)) {
+        sorters = sorters.split(',')
+    }
+
+    return sorters
+        .map((sorter) => {
+            const [column, order] = sorter.split('_')
+            if (!column || !order || !SHORT_TO_FULL_ORDERS_MAP[order]) return
+            return { columnKey: column, order: SHORT_TO_FULL_ORDERS_MAP[order] }
+        })
+        .filter(Boolean)
+}
+
+export const convertSortersToSortBy = (sorters?: SorterColumn | Array<SorterColumn>): Array<string> => {
+    if (!sorters) {
+        return []
+    }
+    if (!Array.isArray(sorters)) {
+        sorters = [sorters]
+    }
+    return sorters
+        .map((sorter) => {
+            if (!FULL_TO_SHORT_ORDERS_MAP[sorter.order]) return
+            return `${sorter.columnKey}_${FULL_TO_SHORT_ORDERS_MAP[sorter.order]}`
+        })
+        .filter(Boolean)
+}
+
+export const getFiltersFromQuery = (query: ParsedUrlQuery): { [x: string]: QueryArgType } => {
+    const { filters } = query
+    if (!filters || typeof filters !== 'string') {
+        return {}
+    }
+    try {
+        const json = JSON.parse(filters)
+        const result = {}
+        Object.keys(json).forEach((key) => {
+            const value = json[key]
+            if (Array.isArray(value)) {
+                result[key] = value.filter((v) => typeof v === 'string' && !!v)
+            } else {
+                if (typeof value === 'string' && !!value) result[key] = value
+            }
+        })
+        return result
+    } catch (e) {
+        return {}
+    }
+}
+
+export const getPageIndexFromOffset = (offset: number, pageSize: number): number => {
+    return Math.floor(offset / pageSize) + 1
+}
+
+export const parseQuery = (query: ParsedUrlQuery): ParsedQueryType => {
+    const filters = getFiltersFromQuery(query)
+    const sorters = getSortersFromQuery(query)
+    const queryOffset = get(query, 'offset', '0')
+    const offset = Number(queryOffset) ? Number(queryOffset) : 0
+    return { filters, sorters, offset }
+}
+
+export const convertColumns = (
+    columns: Array<ColumnInfo>,
+    filters: { [x: string]: QueryArgType },
```

Comment:
move to types, add readable variabale names

Reply:
done 👍